mm2src/coins/rpc_command/lightning/open_channel.rs

```diff
@@ -0,0 +1,237 @@
+    #[display(fmt = "Lightning network is not supported for {}", _0)]
+    UnsupportedCoin(String),
+    #[display(fmt = "Balance Error {}", _0)]
+    BalanceError(String),
+    #[display(fmt = "Invalid path: {}", _0)]
+    InvalidPath(String),
+    #[display(fmt = "Failure to open channel with node {}: {}", _0, _1)]
+    FailureToOpenChannel(String, String),
+    #[display(fmt = "RPC error {}", _0)]
+    RpcError(String),
+    #[display(fmt = "Internal error: {}", _0)]
+    InternalError(String),
+    #[display(fmt = "I/O error {}", _0)]
+    IOError(String),
+    #[display(fmt = "DB error {}", _0)]
+    DbError(String),
+    ConnectToNodeError(String),
+    #[display(fmt = "No such coin {}", _0)]
+    NoSuchCoin(String),
+    #[display(fmt = "Generate Tx Error {}", _0)]
+    GenerateTxErr(String),
+}
+
+impl HttpStatusCode for OpenChannelError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            OpenChannelError::UnsupportedCoin(_) | OpenChannelError::RpcError(_) => StatusCode::BAD_REQUEST,
+            OpenChannelError::FailureToOpenChannel(_, _)
+            | OpenChannelError::ConnectToNodeError(_)
+            | OpenChannelError::InternalError(_)
+            | OpenChannelError::GenerateTxErr(_)
+            | OpenChannelError::IOError(_)
+            | OpenChannelError::DbError(_)
+            | OpenChannelError::InvalidPath(_)
+            | OpenChannelError::BalanceError(_) => StatusCode::INTERNAL_SERVER_ERROR,
+            OpenChannelError::NoSuchCoin(_) => StatusCode::NOT_FOUND,
+        }
+    }
+}
+
+impl From<ConnectionError> for OpenChannelError {
+    fn from(err: ConnectionError) -> OpenChannelError { OpenChannelError::ConnectToNodeError(err.to_string()) }
+}
+
+impl From<CoinFindError> for OpenChannelError {
+    fn from(e: CoinFindError) -> Self {
+        match e {
+            CoinFindError::NoSuchCoin { coin } => OpenChannelError::NoSuchCoin(coin),
+        }
+    }
+}
+
+impl From<BalanceError> for OpenChannelError {
+    fn from(e: BalanceError) -> Self { OpenChannelError::BalanceError(e.to_string()) }
+}
+
+impl From<NumConversError> for OpenChannelError {
+    fn from(e: NumConversError) -> Self { OpenChannelError::InternalError(e.to_string()) }
+}
+
+impl From<GenerateTxError> for OpenChannelError {
+    fn from(e: GenerateTxError) -> Self { OpenChannelError::GenerateTxErr(e.to_string()) }
+}
+
+impl From<UtxoRpcError> for OpenChannelError {
+    fn from(e: UtxoRpcError) -> Self { OpenChannelError::RpcError(e.to_string()) }
+}
+
+impl From<UnexpectedDerivationMethod> for OpenChannelError {
+    fn from(e: UnexpectedDerivationMethod) -> Self { OpenChannelError::InternalError(e.to_string()) }
+}
+
+impl From<std::io::Error> for OpenChannelError {
+    fn from(err: std::io::Error) -> OpenChannelError { OpenChannelError::IOError(err.to_string()) }
+}
+
+impl From<SqlError> for OpenChannelError {
+    fn from(err: SqlError) -> OpenChannelError { OpenChannelError::DbError(err.to_string()) }
+}
+
```

Comment:
the proc macro for deriving enum from variant I wrote could be very useful in this case to write less code

Reply:
Yes, It should be useful in reducing the code :), but this is not in dev branch yet
You can open an issue describing how this will be useful and label it with `refactor`/`Low priority`, or you can add this note as a comment in the errors refactoring issue if this will be used in error enums only

Reply:
ok thanks